Chore/#130341337/convert from grunt to gulp taskrunner

cyrielo opened the pull request:
#### What does this PR do?

This PR converts tasks from grunt task to gulp task using the corresponding gulp plugins for the tasks.
#### Description of Task to be completed?

This task was to convert all grunt tasks to gulp
#### How should this be manually tested?

This can be tested by preceding the task name with gulp instead of grunt
#### Any background context you want to provide?

None
#### What are the relevant pivotal tracker stories?

The relevant story is #130341337
#### Screenshots (if appropriate)

None

demiladebamgbose committed:
set up karma and add base test

Commit:
Converts some grunt tasks to gulp, update package.json for new dependecies

Commit:
Converts some grunt tasks to gulp, update package.json for new dependecies

Commit:
Converts some grunt tasks to gulp, update package.json for new dependecies

demiladebamgbose committed:
change test command to karma start

demiladebamgbose committed:
add test for an angular controller

demiladebamgbose committed:
fix indentation

Commit:
Add fix for mongodb connection

Commit:
Add fix for mongodb connection

Commit:
Add fix for mongodb connection

Commit:
chore(continuous-integration) : adds ci badge to readme, adds travis config

Commit:
chore(continuous-integration) : adds ci badge to readme, adds travis config

Commit:
chore(continuous-integration) : adds ci badge to readme, adds travis config

Commit:
chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

Commit:
chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

Commit:
chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

Commit:
Removed grunt file and traces of grunt dependencies in package.json

Commit:
Removed grunt file and traces of grunt dependencies in package.json

Commit:
Removed grunt file and traces of grunt dependencies in package.json

Commit:
chore(continuous-integration) : adds houndci config

Commit:
chore(continuous-integration) : adds houndci config

demiladebamgbose committed:
refactor karma and jasmine setup

fix indentation and length of lines

finishes
g

ruqoyyasadiq commented:
Follow the PR naming and description convention as stated in the wiki.

demiladebamgbose committed:
refactor fix indentation

indent one space after line break

finishes

ruqoyyasadiq committed:
fix indentation

ruqoyyasadiq committed:
Merge pull request #1 from andela/chore/#130341335/jasmine-and-karma-setup

#130341335 jasmine and karma setup

ruqoyyasadiq reviewed (changes requested):
Fix conflict, ensure to update with develop and most important, follow the PR convention for naming and description :)

Commit:
Merge branch 'chore/#130341331/integrate-travis-ci' of https://github.com/andela/project-kinsan into chore/#130341331/integrate-travis-ci

ruqoyyasadiq committed:
Merge pull request #2 from andela/chore/#130341331/integrate-travis-ci

#130341331 integrate-travis-ci

Commit:
Fix broken lib files links in

Commit:
Fix broken lib files links in

Commit:
Merge branch 'chore/#130341333/integrate-houndci' of https://github.com/andela/project-kinsan into chore/#130341333/integrate-houndci

Commit:
chore (continous-integration) : Fix hound ci issues.

Commit:
chore (continous-integration) : Fix hound ci issues.

ruqoyyasadiq committed:
Merge branch 'develop' into chore/#130341333/integrate-houndci

Commit:
Adds karma task to gulp tasks

ruqoyyasadiq committed:
Merge pull request #3 from andela/chore/#130341333/integrate-houndci

#130341333 integrate-houndci

Commit:
Fix conflict between local and remote branch

Commit:
#130373385 readme-update (#4)

* set up karma and add base test

* change test command to karma start

* add test for an angular controller

* fix indentation

* chore(continuous-integration) : adds ci badge to readme, adds travis config

* chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

* chore(continuous-integration) : adds houndci config

* chore(readme-update): update read me with new relevant info
Finishes #130373385

* refactor karma and jasmine setup

fix indentation and length of lines

finishes
g

* refactor fix indentation

indent one space after line break

finishes

* fix indentation

* chore(continuous-integration) : adds ci badge to readme, adds travis config

* chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

* chore(continuous-integration) : adds ci badge to readme, adds travis config

* chore(continuous-integration) : adds src folder to nodemon watch,  runs only test on travis

* chore(continuous-integration) : adds houndci config

* chore (continous-integration) : Fix hound ci issues.

* chore (continous-integration) : Fix hound ci issues.

* chore(readme-update): update read me with new relevant info
Finishes #130373385

ruqoyyasadiq commented on server.js:
```diff
@@ -5,13 +5,14 @@ var express = require('express'),
+dotenv.config();
 //Load configurations
 //if test env, load example file
 var env = process.env.NODE_ENV = process.env.NODE_ENV || 'development',
```
use ternary operator `process.env.NODE_ENV ? process.env.NODE_ENV : 'development'`

Commit:
Update unresolved conflicts and fixes houndci reported errors

Commit:
Update unresolved conflicts and fixes houndci reported errors

ruqoyyasadiq committed:
resolve conflict

ruqoyyasadiq committed:
Merge branch 'chore/#130341337/convert-from-Grunt-to-Gulp-taskrunner' of github.com:andela/project-kinsan into chore/#130341337/convert-from-Grunt-to-Gulp-taskrunner
